url.bs

```diff
@@ -88,20 +88,30 @@ valid input. User agents, especially conformance checkers, are encouraged to rep
 
 <h3 id=parsers>Parsers</h3>
 
-<p>The <dfn>EOF code point</dfn> is a conceptual code point that signifies the end of a
-string or code point stream.
+<p>The <dfn>EOF code point</dfn> is a conceptual code point that signifies the end of a string or
+code point stream.
 
-<p>Within a parser algorithm that uses a <var>pointer</var> variable, <dfn>c</dfn>
-references the code point the <var>pointer</var> variable points to.
+<p>A <dfn>pointer</dfn> for a <a for=/>string</a> <var>input</var> is an integer that points to a
+<a for=/>code point</a> within <var>input</var>. Initially it points to the start of
+<var>input</var>. If it is &minus;1 it points nowhere. If it is greater than or equal to
+<var>input</var>'s <a for=string>code point length</a>, it points to the <a>EOF code point</a>.
 
-<p>Within a string-based parser algorithm that uses a <var>pointer</var> variable,
-<dfn>remaining</dfn> references the substring after <var>pointer</var> in the string
-being processed.
+<p>When a <a>pointer</a> is defined, <dfn>c</dfn> references the <a for=/>code point</a>
+<a>pointer</a> points to as long as it does not point nowhere. When <a>pointer</a> points to nowhere
+<a>c</a> cannot be used.
+
+<p>When a <a>pointer</a> is defined, <dfn>remaining</dfn> references the substring after <a>c</a> as
+long as <a>c</a> is not the <a>EOF code point</a>. When <a>c</a> is the <a>EOF code point</a>
+<a>remaining</a> cannot be used.
 
 <p class=example id=example-12672b6a>If "<code>mailto:username@example</code>" is a <a>string</a>
-being processed and <var>pointer</var> points to @, <a>c</a> is U+0040 (@) and <a>remaining</a> is
+being processed and <a>pointer</a> points to @, <a>c</a> is U+0040 (@) and <a>remaining</a> is
```

Comment:
"pointer points to" seems a bit strange; should it be "a pointer" or "the pointer"?

Reply:
I can make it "a pointer"